spring-core/src/main/java/facade/BookingFacadeImpl.java

```diff
@@ -0,0 +1,149 @@
+        LOG.debug("Events: " + events);
+        return events.isEmpty() ? Collections.emptyList() : listByPage(events, pageSize, pageNum);
+    }
+
+    public List<Event> getEventsForDay(Date day, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<Event> events = eventService.retrieveEventByDay(day);
+        LOG.debug("Events: " + events);
+        return events.isEmpty() ? Collections.emptyList() : listByPage(events, pageSize, pageNum);
+    }
+
+    public Event createEvent(Event event) {
+        return eventService.addEvent(event);
+    }
+
+    public Event updateEvent(Event event) {
+        return eventService.updateEvent(event);
+    }
+
+    public boolean deleteEvent(long eventId) {
+        return eventService.removeEvent(eventId);
+    }
+
+    public User getUserById(long userId) {
+        return userService.retrieveUserById(userId);
+    }
+
+    public User getUserByEmail(String email) {
+        return userService.retrieveUserByEmail(email);
+    }
+
+    public List<User> getUsersByName(String name, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        final List<User> users = userService.retrieveUserByName(name);
+        LOG.debug("Users: " + users);
+        return users.isEmpty() ? Collections.emptyList() : listByPage(users, pageSize, pageNum);
+    }
+
+    public User createUser(User user) {
+        return userService.addUser(user);
+    }
+
+    public User updateUser(User user) {
+        return userService.updateUser(user);
+    }
+
+    public boolean deleteUser(long userId) {
+        return userService.removeUser(userId);
+    }
+
+    public Ticket bookTicket(long userId, long eventId, int place, Ticket.Category category) {
+        List<Ticket> bookedTickets = getBookedTickets(eventId);
+        final Optional<Ticket> bookedTicket = bookedTickets.stream()
+                .filter(ticket -> place == ticket.getPlace() && category.equals(ticket.getCategory()) && userId == ticket.getUserId()).findFirst();
+        if (bookedTicket.isPresent()) {
+            LOG.error("User is already preset: " + bookedTicket.get());
+            throw new IllegalStateException("Place already booked");
+        }
+        Ticket ticket = new Ticket();
+        ticket.setUserId(userId);
+        ticket.setEventId(eventId);
+        ticket.setPlace(place);
+        ticket.setCategory(category);
+        return ticketService.bookTicket(ticket);
+    }
+
+    public List<Ticket> getBookedTickets(User user, int pageSize, int pageNum) {
+        checkPageSize(pageSize, pageNum);
+        List<Ticket> tickets = ticketService.retrieveUserTickets(user.getId());
+        LOG.debug("Tickets: " + tickets);
+        tickets.sort(ticketComparatorByEventDate().reversed());
+        LOG.debug("Sorted tickets: " + tickets);
+        return listByPage(tickets, pageSize, pageNum);
+    }
+
+    public List<Ticket> getBookedTickets(Event event, int pageSize, int pageNum) {
```

Comment:
method getBookingTickets takes argument Event, but inside this method we have using only event.getId(), you need provide elements only that you used

Reply:
The interface was provided by task creator, I only implemented this method.